src/main/java/com/book/store/annotation/FieldMatchValidator.java

```diff
@@ -0,0 +1,38 @@
+package com.book.store.annotation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Objects;
+import org.apache.commons.beanutils.BeanUtils;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+    private String firstFieldName;
+    private String secondFieldName;
+    private String message;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        this.firstFieldName = constraintAnnotation.first();
+        this.secondFieldName = constraintAnnotation.second();
+        this.message = constraintAnnotation.message();
+    }
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext context) {
+        try {
+            Object firstValue = BeanUtils.getProperty(value, firstFieldName);
+            Object secondValue = BeanUtils.getProperty(value, secondFieldName);
+            boolean valid = Objects.equals(firstValue, secondValue);
+            if (!valid) {
+                context.disableDefaultConstraintViolation();
+                context.buildConstraintViolationWithTemplate(message)
+                        .addPropertyNode(secondFieldName)
+                        .addConstraintViolation();
+            }
+            return valid;
+        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
```

Comment:
all these exceptions extend ReflectiveOperationException
```suggestion
        } catch (ReflectiveOperationException e) {
```